src/components/sidebar/right-sidebar/datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useContext, useEffect, useState } from "react";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import { Input } from "semantic-ui-react";
+import { AppContext } from "../../../../context";
+import httpClient from "../../../../httpClient";
+import { Filters, FilterType, PageState, State } from '../../../../types';
+import { updateFilters } from "../../../../utils/stateUpdateUtils";
+import Translate from "../../../../utils/translate/translateService";
+
+interface DatepickerProps {
+  page: string
+}
+
+export default function Datepicker({ page }: DatepickerProps) {
+  const [state, dispatch] = useContext(AppContext);
+  const pageState = state[page as keyof State] as PageState;
+  const [filterStartDate, filterEndDate] = Array.from(pageState.filters.publish_date);
+  const [startDate, setStartDate] = useState(filterStartDate);
+  const [endDate, setEndDate] = useState(filterEndDate);
+
+  const datePickerChanged = async (isStart: Boolean, date: Date) => {
+    const newDates = [startDate, endDate];
+    newDates[0] = isStart ? date : startDate;
+    newDates[1] = !isStart ? date : endDate;
+
+    if (isStart) {
+      setStartDate(date);
+    }
+    else {
+      setEndDate(date);
+    }
+    await updateFilters(dispatch,
+      pageState.filters,
+      "publish_date" as FilterType,

```

Comment:
I think we're filtering by sampling_end_date now?

Reply:
Don't think this nomenclature affects anything (since the backend is responsible for the actual filtering), but maybe we want to rename for consistency

Reply:
We can do that yeah, just didn't worry about it for now